File: 1cpp/Pentagon.cpp
//
// Created by olile on 10.04.24.
//

#include "Pentagon.h"

#include <cmath>
#include <stdexcept>

Pentagon::Pentagon(double a) {
    if(a < 0){
        throw std::invalid_argument("żaden argument nie może być  ujemny.");
    }
    this->a = a;
}

double Pentagon::calcArea(){
    return sqrt(5*(5+2*sqrt(5)))*this->a*this->a/4;
}

double Pentagon::calcCircumference(){
    return this->a*5;
}

std::string Pentagon::info(){
    return "Pentagon";
}

File: 1java/1java/src/Pentagon.java
// Pentagon.java (Example implementation)
public class Pentagon implements Shape {
    private double a;

    public Pentagon(double a) {
        if (a < 0) {
            throw new IllegalArgumentException("Argument must not be negative.");
        }
        this.a = a;
    }

    @Override
    public double calcArea() {
        // Implement the area calculation for a regular pentagon
        // Return the actual area value.
        return Math.sqrt(5 * (5 + 2 * Math.sqrt(5))) * a * a / 4;
    }

    @Override
    public double calcCircumference() {
        return a * 5;
    }

    @Override
    public String info() {
        return "Pentagon";
    }
}
